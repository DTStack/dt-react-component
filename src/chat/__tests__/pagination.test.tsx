import React from 'react';
import { act } from 'react-dom/test-utils';
import { fireEvent, render } from '@testing-library/react';

import Pagination from '../pagination';

describe('Test Chat Pagination', () => {
    it('Match Snapshots', () => {
        expect(
            render(<Pagination current={1} total={5} onChange={jest.fn()} />).asFragment()
        ).toMatchSnapshot('default');

        expect(
            render(<Pagination disabled current={1} total={5} onChange={jest.fn()} />).asFragment()
        ).toMatchSnapshot('disabled');
    });

    it('Should call onChange when click left/right arrow', () => {
        const onChange = jest.fn();
        const { container, rerender } = render(
            <Pagination current={1} total={5} onChange={onChange} />
        );

        act(() => {
            fireEvent.click(container.querySelector('.anticon-left')!);
        });
        expect(onChange).not.toBeCalled();

        act(() => {
            fireEvent.click(container.querySelector('.anticon-right')!);
        });
        expect(onChange).toBeCalledWith(2);

        rerender(<Pagination current={5} total={5} onChange={onChange} />);

        act(() => {
            fireEvent.click(container.querySelector('.anticon-left')!);
        });
        expect(onChange).toBeCalledWith(4);

        onChange.mockRestore();
        act(() => {
            fireEvent.click(container.querySelector('.anticon-right')!);
        });
        expect(onChange).not.toBeCalled();
    });
});
