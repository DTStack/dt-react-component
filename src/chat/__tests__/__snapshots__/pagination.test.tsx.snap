// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Chat Pagination Match Snapshots: default 1`] = `
<DocumentFragment>
  <div
    class="dtc-aigc-pagination"
  >
    <span
      class="arrow-disabled"
      data-mock-icon="LeftOutlined"
      data-testid="icon-left"
    />
    <section>
      <span>
        1
      </span>
      <span>
        /
      </span>
      <span>
        5
      </span>
    </section>
    <span
      class=""
      data-mock-icon="RightOutlined"
      data-testid="icon-right"
    />
  </div>
</DocumentFragment>
`;

exports[`Test Chat Pagination Match Snapshots: disabled 1`] = `
<DocumentFragment>
  <div
    class="dtc-aigc-pagination"
  >
    <span
      class="arrow-disabled"
      data-mock-icon="LeftOutlined"
      data-testid="icon-left"
    />
    <section>
      <span>
        1
      </span>
      <span>
        /
      </span>
      <span>
        5
      </span>
    </section>
    <span
      class="arrow-disabled"
      data-mock-icon="RightOutlined"
      data-testid="icon-right"
    />
  </div>
</DocumentFragment>
`;
